src/components/SaveAsTemplateModal.tsx

```diff
@@ -0,0 +1,115 @@
+import ReactModal from "react-modal";
+import { FormattedMessage } from "react-intl";
+import { useCreatePrivateTemplate } from "../queries/template-queries";
+import {
+  CampaignContent,
+  TemplateContent,
+} from "../abstractions/domain/content";
+
+interface SaveAsTemplateModalProps {
+  onClose: () => void;
+  content: TemplateContent;
+}
+
+export const SaveAsTemplateModal = ({
+  onClose,
+  content,
+}: SaveAsTemplateModalProps) => {
+  const {
+    mutate: createPrivateTemplate,
+    isLoading,
+    isSuccess,
+  } = useCreatePrivateTemplate();
+
+  const createTemplate = () => {
+    createPrivateTemplate(content);
+  };
+
+  const successContent = (
+    <>
+      <p>
+        <FormattedMessage id="new_template_has_been_saved"></FormattedMessage>
+      </p>
+      <div className="form-request">
+        <div className="container-buttons">
+          <button
+            type="button"
+            name="ok"
+            className="dp-button button-medium primary-green"
+            onClick={onClose}
+          >
+            <FormattedMessage id="accept"></FormattedMessage>
+          </button>
+        </div>
+      </div>
+    </>
+  );
+
+  const formContent = (
+    <>
+      <p>
+        <FormattedMessage id="new_template_description"></FormattedMessage>
+      </p>
+      <form action="#" className="form-request">
```

Comment:
Is using `action=#` good practice in forms?

Reply:
I did change this for a Controlled Form, now we send the request to create template from onSubmit method

Reply:
👍